src/components/PrivacyModal/PrivacyModal.hooks.tsx

```diff
@@ -0,0 +1,59 @@
+import { SnackbarContext } from '@services'
+import { useCallback, useContext, useMemo } from 'react'
+import { useTranslation } from 'react-i18next'
+import { useCookies } from 'react-cookie'
+
+/**
+ * @prop privacyPolicyCookieSet - Currently set cookie
+ * @prop handleAccept - sets the cookie and displays a Snackbar
+ * @category Hooks
+ */
+
+export type PrivacyModalHookReturn = {
+  readonly privacyPolicyCookie: boolean
+  readonly handleAccept: (isAccepted: boolean) => void
+}
+
+/**
+ * Hook for the PrivacyModal logic.
+ * Provides function for setting the cookie and a prop that returns the cookie.
+ * @returns - cookie and handleAccept function.
+ * @category Hooks
+ */
+
+export const usePrivacyModal = (): PrivacyModalHookReturn => {
+  const { t } = useTranslation()
+  const { addSnackbar } = useContext(SnackbarContext)
+  const [cookies, setCookie] = useCookies(['privacy_accept_token'])
+
+  //**Logic **//
+  const handleAccept = useCallback(
+    (isAccepted: boolean) => {
+      if (isAccepted) {
+        setCookie('privacy_accept_token', true, { path: '/' })
+        addSnackbar({
+          message: t('components.PrivacyModal.accepted'),
+          severity: 'success'
+        })
+      } else {
+        //if decline should set a cookie with expiration uncomment
+        //const tomorrow = new Date()
+        //tomorrow.setDate(tomorrow.getDate() + 1)
+        //setCookie('privacy_accept_token', false, { path: '/', expires: tomorrow })
+        addSnackbar({
+          message: t('components.PrivacyModal.declined'),
+          severity: 'warning'
+        })
+      }
+    },
+    [addSnackbar, setCookie]
+  )
+  const privacyPolicyCookie = cookies['privacy_accept_token'] != null
```

Comment:
Wenn spart sich eine logische Überprüfung, wenn man direkt den Cookie returned, um ihn in der Komponente direkt zu verwenden.

Also statt hier einen Boolean zu erzeugen, indem überprüft wird, ob der Cookie existert, um ihn dann später auf true oder false zu überprüfen, kann man auch direkt den Cookie returnen, um die Existenz des Cookies als true/false für das Rendering der Komponente verwenden.

```suggestion
```